htdocs/blockedlog/class/blockedlog.class.php

```diff
@@ -478,26 +478,30 @@ public function setObjectData(&$object, $action, $amounts, $fuser = null)
 			$object->fetch_thirdparty();
 		}
 		if (!empty($object->thirdparty)) {
-			$this->object_data->thirdparty = new stdClass();
+			$thirdparty = $object->thirdparty;
 
-			foreach ($object->thirdparty as $key => $value) {
-				if (in_array($key, $arrayoffieldstoexclude)) {
-					continue; // Discard some properties
-				}
-				if (!in_array($key, array(
-				'name', 'name_alias', 'ref_ext', 'address', 'zip', 'town', 'state_code', 'country_code', 'idprof1', 'idprof2', 'idprof3', 'idprof4', 'idprof5', 'idprof6', 'phone', 'fax', 'email', 'barcode',
-				'tva_intra', 'localtax1_assuj', 'localtax1_value', 'localtax2_assuj', 'localtax2_value', 'managers', 'capital', 'typent_code', 'forme_juridique_code', 'code_client', 'code_fournisseur'
-				))) {
-					continue; // Discard if not into a dedicated list
-				}
-				if (!is_object($value) && !is_null($value) && $value !== '') {
-					$this->object_data->thirdparty->$key = $value;
+			if (is_array($thirdparty) || ($thirdparty instanceof Traversable)) {
+				$this->object_data->thirdparty = new stdClass();
+
+				foreach ($thirdparty as $key => $value) {
+					if (in_array($key, $arrayoffieldstoexclude)) {
+						continue; // Discard some properties
+					}
+					if (!in_array($key, array(
+					'name', 'name_alias', 'ref_ext', 'address', 'zip', 'town', 'state_code', 'country_code', 'idprof1', 'idprof2', 'idprof3', 'idprof4', 'idprof5', 'idprof6', 'phone', 'fax', 'email', 'barcode',
+					'tva_intra', 'localtax1_assuj', 'localtax1_value', 'localtax2_assuj', 'localtax2_value', 'managers', 'capital', 'typent_code', 'forme_juridique_code', 'code_client', 'code_fournisseur'
+					))) {
+						continue; // Discard if not into a dedicated list
+					}
+					if (!is_object($value) && !is_null($value) && $value !== '') {
+						$this->object_data->thirdparty->$key = $value;
+					}
 				}
 			}
 		}
 
 		// Add company info
-		if (!empty($mysoc)) {
+		if (!empty($mysoc) && (is_array($mysoc) || ($mysoc instanceof Traversable))) {
```

Comment:
It seems that a PHP object will return false with 
$object instanceof Traversable.
And we want to do the foreach($object as $key => $value) to loop on each of its property.
So the test will break this.

Reply:
Ok, I assumed that this was not possible but it is.

So I suppress the notice.